test/execution_summary_test.py

```diff
@@ -1086,3 +1088,87 @@ class Bar(RunOnceTask):
         self.assertIn('Luigi Execution Summary', s)
         self.assertNotIn('00:00:00', s)
         self.assertNotIn('\n\n\n', s)
+
+class ExecutionSummaryWithTaskRetryTest(LuigiTestCase):
```

Comment:
Do you really need a whole new class? Is it not possible to copy any of the tests above and just slightly modify them to test what you're trying?

In particular I don't see why you need to do writes to the file system and `os.sleep()`.